clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -172,73 +172,58 @@ public PollResult poll(long currentTimeMs) {
                         k -> sessionHandlers.computeIfAbsent(node.id(), n -> new ShareSessionHandler(logContext, n, memberId)));
 
                 TopicIdPartition tip = new TopicIdPartition(topicId, partition);
-                Acknowledgements acknowledgementsToSend = fetchAcknowledgementsToSend.remove(tip);
-                if (acknowledgementsToSend != null) {
-                    metricsManager.recordAcknowledgementSent(acknowledgementsToSend.size());
-                    fetchAcknowledgementsInFlight.put(tip, acknowledgementsToSend);
+                Acknowledgements acknowledgementsToSend = null;
+                Map<TopicIdPartition, Acknowledgements> nodeAcknowledgementsMap = fetchAcknowledgementsToSend.get(node.id());
```

Comment:
Query: Can there be ever some acknowledgments which can starve in the `fetchAcknowledgementsToSend` when some nodeId is never received from broker?

Reply:
I think there will need to be a follow-on PR for this. In a way, you're correct, but I think to maintain the ordering of things, we need to fail some acknowledgements more aggressively from the poll loop. I had been avoiding that, but I think it's inevitable actually.

Reply:
Thank you.